### ISSUE TITLE
Filter `Missing value` errors from austraits$excluded_data

### ISSUE THREAD

**Issue by ehwenk** [traitecoevo/austraits.build#539]
There are currently 811410 `Missing value` errors in `austraits$excluded_data`:
```
austraits$excluded_data %>% filter(error == "Missing value") %>% nrow()
```
These entirely represent blank cells in wide datasets that are legitimately blank. They include instances of:
* filtering out duplicate species-level categorical trait observations
* filtering out duplicates
* where a data.csv file was created by binding together two spreadsheets with measurements on different individuals or traits, leading to large numbers of blank cells

The vast number dwarfs the 3579 other values in the `austraits$excluded_data` table, making these harder to search and, for anyone other than someone on the AusTraits team will make the  `austraits$excluded_data` table almost pointless to explore.

I recommend filtering out all `Missing value` errors as the default. We can include an if_else loop, that allows these to be included, if we want to check them during the actual inputting of a study.

**Comment by GaryTruong** [comment]
This is a relatively simple fix adding the extra line to filter out "Missing value" from the error column in `load_study`. I've added the extra line on a separate branch. 

Adding the loop may not be needed as you can run this part of load_study to generate the error column containing the missing values and then filter by "Missing values" or non NA values. But can be added if you think it is necessary.

```
  # load and clean trait data
  traits <- readr::read_csv(filename_data_raw, col_types = cols(), guess_max = 100000, progress=FALSE) %>%
    custom_manipulation(metadata[["config"]][["custom_R_code"]])() %>%
    parse_data(dataset_id, metadata) %>%
    add_all_columns(definitions$columns_traits) %>%
    flag_unsupported_traits(definitions) %>%
    flag_excluded_observations(metadata) %>%
    convert_units(definitions, unit_conversion_functions) %>%
    flag_unsupported_values(definitions) %>%
    apply_taxonomic_updates(metadata) %>%
    dplyr::mutate(
      # For cells with multiple values (separated by a space), sort these alphabetically
      value = ifelse(is.na(.data$error), split_then_sort(.data$value), .data$value),
      value_type = factor(.data$value_type, levels = names(definitions$value_type$values)),
      #ensure dates are converted back to character
      date = as.character(.data$date)
      ) %>%
    dplyr::arrange(.data$observation_id, .data$trait_name, .data$value_type)
```

**Comment by ehwenk** [comment]
My reason for excluding the `Missing Values` is that they are meaningless in completed studies, but they absolutely overwhelm that number of trait values excluded for other purposes and therefore other people looking at `austraits$excluded_data` will be unlikely to focus in one the trait values that have been excluded due to an actual error.

So my thought was that `Missing Values` would be excluded as the default, but there would be some way to see them, since while inputting a study there are times you want to check and ensure data aren't inappropriately landing in `Missing Values`. (These are rare occasions - I'd rather they were filtered out altogether, than had to be manually filtered out each time.)

**Comment by GaryTruong** [comment]
Yeah 100% agree that they should be filtered out. What do you think about adding a new function to do this in potentially support.R or maybe one of the report.R files and you can call that function to check the missing values. That way we can leave the main excluded_data tables without `Missing values`.

**Comment by ehwenk** [comment]
Yes, I like that idea. Probably in `support.R`, since it won't actually be a function to build the report.

**Comment by GaryTruong** [comment]
Rather than add a new fuction, add an additional argument to load_study with a default to remove NAs from excluded data table with an if statement. Default remove missing_values, alternative is keep missing_values.

**Comment by dfalster** [comment]
Closed via #578
